contracts/external/cw-fund-distributor/src/contract.rs

```diff
@@ -194,58 +199,45 @@ pub fn execute_claim_cw20s(
         return Err(ContractError::EmptyClaim {});
     }
 
-    let relative_share = get_relative_share(deps.as_ref().borrow(), sender.clone());
-
-    let messages: Vec<WasmMsg> = tokens
-        .into_iter()
-        .map(|addr| {
-            // get the balance of distributor at instantiation
-            let bal = CW20_BALANCES
-                .load(deps.storage, Addr::unchecked(addr.clone()))
-                .unwrap();
-            // check for any previous claims
-            let previous_claim = CW20_CLAIMS
-                .load(
-                    deps.storage,
-                    (sender.clone(), Addr::unchecked(addr.clone())),
-                )
-                .unwrap_or_default();
-
-            // get % share of sender and subtract any previous claims
-            let entitlement = get_entitlement(bal, relative_share, previous_claim);
+    let relative_share = get_relative_share(&deps.as_ref(), sender.clone())?;
+    let mut messages: Vec<WasmMsg> = vec![];
+    for addr in tokens {
+        // get the balance of distributor at instantiation
+        let bal = CW20_BALANCES.load(deps.storage, Addr::unchecked(addr.clone()))?;
+
+        // check for any previous claims
+        let previous_claim = CW20_CLAIMS
+            .load(
+                deps.storage,
+                (sender.clone(), Addr::unchecked(addr.clone())),
+            )
+            .unwrap_or_default();
```

Comment:
it may be better to do `may_load(deps.storage, (sender.clone(), Addr::unchecked(addr.clone())))?.unwrap_or_default()` here so we can distinguish between an error happened while loading, and there was no value set.